pkg/gadgets/bindsnoop/tracer/core/tracer.go

```diff
@@ -0,0 +1,298 @@
+//go:build linux
+// +build linux
+
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019-2022 The Inspektor Gadget authors
+
+package tracer
+
+// #include <linux/types.h>
+// #include "./bpf/bindsnoop.h"
+// #include <arpa/inet.h>
+// #include <stdlib.h>
+//
+//char *ip_to_string(const struct bind_event *event) {
+//	socklen_t size;
+//	int ip_type;
+//	char *ip;
+//
+//	size = sizeof *ip * INET6_ADDRSTRLEN;
+//	ip = malloc(size);
```

Comment:
Can't you just pass `INET6_ADDRSTRLEN` to `malloc()`? Btw I'm not sure that calling `malloc()/free()` has any advantage over using a static buffer and using it as I did in https://github.com/kinvolk/inspektor-gadget/pull/518/commits/c7660a8a9c1dfd7d9c0f257819790691d96408d8#diff-9091bc851c8f5012405e997dbca4673f05de1b2c85ddb41a99d40a121d20a0a8R25

Reply:
I always prefer to give the size of the element multiplied by the number of element, thus it is portable and if the pointer underlying type changes I do not need to change this code.

Reply:
Good one on the threads, I think I'll change my implementation then to avoid this problem.